packages/runtime/src/state/toolkit.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  combineReducers,
+  type ThunkDispatch,
+  type Middleware,
+  type Reducer,
+  type UnknownAction,
+  type MiddlewareAPI,
+  type ConfigureStoreOptions,
+} from '@reduxjs/toolkit'
+
+import { type Action, isKnownAction } from './actions'
+
+import { serializeState } from '../utils/serializeState'
+
+export function actionMiddleware<S, A extends UnknownAction = UnknownAction>(
+  middleware: (
+    api: MiddlewareAPI<ThunkDispatch<S, unknown, A>, S>,
+  ) => (next: (action: unknown) => unknown) => (action: Action) => unknown,
+): Middleware<unknown, S, ThunkDispatch<S, unknown, A>> {
+  return api => next => {
+    const m = middleware(api)(next)
+    return (action: unknown) => {
+      return isKnownAction(action) ? m(action) : next(action)
```

Comment:
Filter out unknown actions in middleware.

Reply:
Would this interfere with [Redux internal actions](https://github.com/reduxjs/redux/blob/master/src/utils/actionTypes.ts)?

Reply:
So this is not a problem for middleware, but filtering out Redux internal actions *is* a problem for reducers, since Redux relies on the reducer returning the initial state slice for populating the state.

Reply:
https://github.com/makeswift/makeswift/pull/1090/files#r2279692559